js/ai/src/model/middleware.ts

```diff
@@ -235,6 +240,171 @@ export function augmentWithContext(
   };
 }
 
+/**
+ * Options for the `retry` middleware.
+ */
+export interface RetryOptions {
+  /**
+   * The maximum number of times to retry a failed request.
+   * @default 3
+   */
+  maxRetries?: number;
+  /**
+   * An array of `StatusName` values that should trigger a retry.
+   * @default ['UNAVAILABLE', 'DEADLINE_EXCEEDED', 'RESOURCE_EXHAUSTED', 'ABORTED', 'INTERNAL']
+   */
+  statuses?: StatusName[];
+  /**
+   * The initial delay between retries, in milliseconds.
+   * @default 0
```

Comment:
I don't think this should be 0. Maybe 1000 (1 second)? Because if we leave it as everything default, it will retry 3 times immediately with no backoff and no delay which is not as helpful as it could be. Granted a working query might take a while, but for error like resource exhausted it will be very quick and 3 immediate retires won't help.

Reply:
I believe no delay is the correct behaviour by default...